File: binary_search.py
def binary_search(arr,value):
    size=len(arr)
    start=0
    end=size-1

    while(start<=end):
        mid=(start+end)//2
        if(arr[mid]==value):
            return mid
        elif (arr[mid]>value):
            end=mid-1
        elif (arr[mid]<value):
            start=mid+1

    return -1

arr=1,2,7,8,9,12,32,34

res=binary_search(arr,90)
print(res)

File: binary_search.js
const binary_search=(arr,value)=>{
    let start=0;
    let end=arr.length-1;

    while(start<=end){
        let mid=Math.floor((start+end)/2);
        if(arr[mid]===value){
            return mid;
        }else if(arr[mid]<value){
            start=mid+1;
        }else{
            end=mid-1;
        }
    }
    return -1;

}
const arr=[1,2,3,5,6,8,10]

binary_search(arr,8)

